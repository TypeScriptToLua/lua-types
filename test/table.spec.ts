import { describeForEachLuaTarget, tstl } from "./test-utils";

describeForEachLuaTarget("table", target => {

    test("table.concat", () => {
        const lua = tstl(target, `
            const myTable = [1, 2, 3];
            table.concat(myTable, ",", 0, myTable.length);
        `);

        expect(lua).toMatchSnapshot();
    });

    test("table.insert", () => {
        const lua = tstl(target, `
            const myTable = [1, 2, 3];
            table.insert(myTable, 4);
        `);

        expect(lua).toMatchSnapshot();
    });

    test("table.insert at position", () => {
        const lua = tstl(target, `
            const myTable = [1, 2, 3];
            table.insert(myTable, 1, 4);
        `);

        expect(lua).toMatchSnapshot();
    });

    test("table.remove", () => {
        const lua = tstl(target, `
            const myTable = [1, 2, 3];
            table.remove(myTable, 2);
        `);

        expect(lua).toMatchSnapshot();
    });

    test("table.sort", () => {
        const lua = tstl(target, `
            const myTable = [1, 2, 3];
            table.sort(myTable, (a, b) => a < b);
        `);

        expect(lua).toMatchSnapshot();
    });
});
