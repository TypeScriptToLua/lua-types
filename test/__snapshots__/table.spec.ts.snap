// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lua version 5.1 / table table.concat 1`] = `
"myTable = {1, 2, 3}
table.concat(myTable, \\",\\", 0, #myTable)"
`;

exports[`Lua version 5.1 / table table.insert 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 4)"
`;

exports[`Lua version 5.1 / table table.insert at position 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 1, 4)"
`;

exports[`Lua version 5.1 / table table.remove 1`] = `
"myTable = {1, 2, 3}
table.remove(myTable, 2)"
`;

exports[`Lua version 5.1 / table table.sort 1`] = `
"myTable = {1, 2, 3}
table.sort(
    myTable,
    function(a, b) return a < b end
)"
`;

exports[`Lua version 5.2 / table table.concat 1`] = `
"myTable = {1, 2, 3}
table.concat(myTable, \\",\\", 0, #myTable)"
`;

exports[`Lua version 5.2 / table table.insert 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 4)"
`;

exports[`Lua version 5.2 / table table.insert at position 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 1, 4)"
`;

exports[`Lua version 5.2 / table table.remove 1`] = `
"myTable = {1, 2, 3}
table.remove(myTable, 2)"
`;

exports[`Lua version 5.2 / table table.sort 1`] = `
"myTable = {1, 2, 3}
table.sort(
    myTable,
    function(a, b) return a < b end
)"
`;

exports[`Lua version 5.3 / table table.concat 1`] = `
"myTable = {1, 2, 3}
table.concat(myTable, \\",\\", 0, #myTable)"
`;

exports[`Lua version 5.3 / table table.insert 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 4)"
`;

exports[`Lua version 5.3 / table table.insert at position 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 1, 4)"
`;

exports[`Lua version 5.3 / table table.remove 1`] = `
"myTable = {1, 2, 3}
table.remove(myTable, 2)"
`;

exports[`Lua version 5.3 / table table.sort 1`] = `
"myTable = {1, 2, 3}
table.sort(
    myTable,
    function(a, b) return a < b end
)"
`;

exports[`Lua version 5.4 / table table.concat 1`] = `
"myTable = {1, 2, 3}
table.concat(myTable, \\",\\", 0, #myTable)"
`;

exports[`Lua version 5.4 / table table.insert 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 4)"
`;

exports[`Lua version 5.4 / table table.insert at position 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 1, 4)"
`;

exports[`Lua version 5.4 / table table.remove 1`] = `
"myTable = {1, 2, 3}
table.remove(myTable, 2)"
`;

exports[`Lua version 5.4 / table table.sort 1`] = `
"myTable = {1, 2, 3}
table.sort(
    myTable,
    function(a, b) return a < b end
)"
`;

exports[`Lua version JIT / table table.concat 1`] = `
"myTable = {1, 2, 3}
table.concat(myTable, \\",\\", 0, #myTable)"
`;

exports[`Lua version JIT / table table.insert 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 4)"
`;

exports[`Lua version JIT / table table.insert at position 1`] = `
"myTable = {1, 2, 3}
table.insert(myTable, 1, 4)"
`;

exports[`Lua version JIT / table table.remove 1`] = `
"myTable = {1, 2, 3}
table.remove(myTable, 2)"
`;

exports[`Lua version JIT / table table.sort 1`] = `
"myTable = {1, 2, 3}
table.sort(
    myTable,
    function(a, b) return a < b end
)"
`;
